pxr/usdValidation/usdShadeValidators/testenv/testUsdShadeValidators.cpp

```diff
@@ -536,22 +514,246 @@ TestUsdShadeEncapsulationRulesValidator()
+    // UsdUVTexture).
+    UsdStageRefPtr usdStage = UsdStage::CreateInMemory();
+    UsdShadeMaterial material = UsdShadeMaterial::Define(usdStage,
+        SdfPath("/RootMaterial"));
+
+    const std::string usdPreviewSurfaceShaderPath =
+        "/RootMaterial/UsdPreviewSurface";
+    UsdShadeShader usdPreviewSurfaceShader = UsdShadeShader::Define(
+        usdStage, SdfPath(usdPreviewSurfaceShaderPath));
+    const TfToken UsdPreviewSurface("UsdPreviewSurface");
+    usdPreviewSurfaceShader.CreateIdAttr(
+        VtValue(UsdPreviewSurface));
+    UsdPrim usdPreviewSurfaceShaderPrim = usdPreviewSurfaceShader.GetPrim();
+
+    UsdShadeShader usdUvTextureShader = UsdShadeShader::Define(
+        usdStage, SdfPath("/RootMaterial/NormalTexture"));
+    const TfToken UsdUVTexture("UsdUVTexture");
+    usdUvTextureShader.CreateIdAttr(VtValue(UsdUVTexture));
+
+    // Add initial valid file and sourceColorSpace input values.
+    std::string textureAssetPath = "./normalMap.jpg";
+    const TfToken File("file");
+    UsdShadeInput fileInput = usdUvTextureShader.CreateInput(
+        File, SdfValueTypeNames->Asset);
+    fileInput.Set(SdfAssetPath(textureAssetPath));
+    const TfToken SourceColorSpace("sourceColorSpace");
+    UsdShadeInput sourceColorSpaceInput = usdUvTextureShader.CreateInput(
+        SourceColorSpace, SdfValueTypeNames->Token);
+    const TfToken Raw("raw");
+    sourceColorSpaceInput.Set(Raw);
+
+    // Connect the output of the UsdUVTexture Shader to the normal of the
+    // UsdPreviewSurface Shader.
+    const TfToken RGB("rgb");
+    usdUvTextureShader.CreateOutput(RGB, SdfValueTypeNames->Float3);
+    const TfToken Normal("normal");
+    UsdShadeInput normalInput = usdPreviewSurfaceShader.CreateInput(
+        Normal, SdfValueTypeNames->Normal3f);
+    normalInput.ConnectToSource(
+        SdfPath("/RootMaterial/NormalTexture.outputs:rgb"));
+
+    // Verify invalid bias & scale error, they are expected to exist but
+    // do not.
+    {
+        const UsdValidationErrorVector errors = validator->Validate(
+    usdPreviewSurfaceShaderPrim);
+        TF_AXIOM(errors.size() == 1u);
+
+        const TfToken expectedErrorIdentifier(
+        "usdShadeValidators:NormalMapTextureValidator.NonCompliantBiasAndScale");
+        const std::string expectedErrorMsg =
+            TfStringPrintf("UsdUVTexture prim <%s> reads 8 bit Normal Map "
+                           "@./normalMap.jpg@, which requires that "
+                           "inputs:scale be set to (2, 2, 2, 1) and "
+                           "inputs:bias be set to (-1, -1, -1, 0) for proper "
+                           "interpretation as per the UsdPreviewSurface and "
+                           "UsdUVTexture docs.",
+                           usdUvTextureShader.GetPath().GetText());
+        ValidatePrimError(errors[0],
+            expectedErrorIdentifier,
+            usdUvTextureShader.GetPath(),
+            expectedErrorMsg);
+    }
```

Comment:
Looks like this one is failing:

Output from the ValidatePrimError seems to say that error identifier is file error and not NonCompliantBiasAndScale:

```
Error: usdShadeValidators:NormalMapTextureValidator.InvalidFile
Expected Error: usdShadeValidators:NormalMapTextureValidator.NonCompliantBiasAndScale
```

Can you make sure the test is passing on your end?

Reply:
Ahh I see, weirdly the normalMap.jpg didn't get synced for me.